### PR TITLE
Implementation of Task "Reference comic by title/alias"

### PR DESCRIPTION
**Pull request by sadoe**
I added functions in bot.py to handle comics requested by title and created a new table in the database to hold pairs of comic titles and comic numbers.

In order to keep the database updated with new comics, I made a new script which currently runs 24/7 and pulls new comics every 24 hrs (I tried doing this on the fly from within the bot, but it takes a lot of time to fully update the database the first time and that causes it to stop replying to comments until it finishes updating so I thought I would make a new script which can be run in parallel with the bot). I know there must be a better way to do this but I thought I should ask your opinion.

The update script takes a couple of hours (for me) to pull all the comics the first time it runs, but once it gets up to the latest comic, there's practically not delay.

I also made some small tweaks here and there.

### CODE REVIEW

**Comment by joeyvanlierop** [comment]
Thanks for the contribution! This looks fantastic, however you will have to give me a few days to double check everything before merging.

**Comment by sadoe** [comment]
Of course! Take your time. I'm still looking for bugs myself.

**Comment by joeyvanlierop** [comment]
I have made a few minor changes to this pull request, however I cannot push them. Would you mind checking off the box that [allows edits from maintainers](https://github.blog/2016-09-07-improving-collaboration-with-forks/)? Thanks

**Comment by sadoe** [comment]
Done.

File: bot/bot.py [hunk 1]
```diff
@@ -119,9 +119,29 @@ def handle_comment(self, comment, strict_match=True):
...
+            if comic is None:
+                continue
+            elif str(comic["num"]) in comic_ids or str(comic["num"]) in seen:     # check if comic is a duplicate
+                continue
+
+            seen.add(str(comic["num"]))
+
             self.database.add_id(comment_id, comic["num"])
+            response = self.format_response(comic)
+            responses.append(response)
```
Comment (sadoe) [inline bot/bot.py]: These lines fix 8a8caa1860828c5c0140a30b8b9efedd4969acd0. The problem was that we were calling **format_response** before adding the new comment in the database, which leads to a **division by zero** error if there are no other comments in the database.

This prevents the error and also produces the correct percentage. Whereas the current implementation in master, doesn't take into account the latest comment when getting total_count from the database.

No need to change anything. I just wanted to point that out.

**Comment by joeyvanlierop** [comment]
Thank you for your fantastic contribution and thank you also for your patience! I didn't intend to take this long to merge, however I have been quite busy trying to adapt to the crazy state that our world is in. Stay healthy!

**Comment by sadoe** [comment]
Sure, don't worry about it! Thank you for your kind words and I hope you're safe too!

**Pull request merged**